flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java

```diff
@@ -407,10 +410,19 @@ public void afterBulk(long executionId, BulkRequest request, BulkResponse respon
 							LOG.error("Failed Elasticsearch item request: {}", itemResponse.getFailureMessage(), failure);
 
 							restStatus = itemResponse.getFailure().getStatus();
+							actionRequest = request.requests().get(i);
 							if (restStatus == null) {
-								failureHandler.onFailure(request.requests().get(i), failure, -1, failureRequestIndexer);
+								if (actionRequest instanceof ActionRequest) {
```

Comment:
I am fine with how it is now. Did you consider just dropping the types of failed requests we do not know, though? 

Anyway none of those should happen as we only ever send ActionRequests.

Reply:
As you say, it should never happen. But if it does, it'll be a lot easier to debug, because there's an exception.